packages/asc-ui/src/components/Radio/Radio.tsx

```diff
@@ -1,27 +1,85 @@
-import * as React from 'react'
-import RadioItemStyle from './RadioItemStyle'
-import RadioLabelStyle from './RadioLabelStyle'
-import RadioGroupStyle from './RadioGroupStyle'
+import React, { useState, useContext, useMemo } from 'react'
+import RadioStyle, {
+  RadioWrapperStyle,
+  RadioCircleStyle,
+  Props,
+} from './RadioStyle'
+import RadioContext from './RadioContext'
+import LabelContext from '../Label/LabelContext'
 
-interface Props {
-  id: string
-  name?: string
-  label: string
-  value?: string
-  defaultChecked?: boolean
-  disabled?: boolean
-}
+const Radio: React.FC<Props & React.HTMLAttributes<HTMLDivElement>> = ({
+  className,
+  onChange,
+  variant,
+  disabled,
+  id,
+  defaultChecked,
+  error: errorProp,
+  ...otherProps
+}) => {
+  const [focus, setFocus] = useState(false)
+  const {
+    setChecked,
+    checked: checkedProp,
+    name,
+    error: errorGroup,
+  } = useContext(RadioContext)
+  const { setActive } = useContext(LabelContext)
+  const checked = checkedProp === id
+  const error = errorProp || errorGroup || false
 
-const Radio = ({ id, value, label, disabled }: Props) => (
-  <RadioItemStyle>
-    <RadioLabelStyle htmlFor={id} disabled={disabled}>
-      {label || value}
-    </RadioLabelStyle>
-  </RadioItemStyle>
-)
+  // Make the label aware of changes in the checked state
+  useMemo(() => {
+    setActive(checked)
+  }, [checked, setActive])
 
-export default Radio
+  // Set checked on defaultChecked
+  if (defaultChecked && !checkedProp) {
+    setChecked(id)
+  }
 
-export const RadioGroup: React.FC<{}> = ({ children, ...otherProps }) => (
-  <RadioGroupStyle {...otherProps}>{children}</RadioGroupStyle>
-)
+  return (
+    <RadioWrapperStyle
+      {...{
+        focus,
+        error,
+        className,
+        variant,
+        disabled,
+        checked,
+      }}
+      aria-disabled={disabled}
+    >
+      <RadioCircleStyle
+        {...{
+          error,
+          disabled,
+          focus,
+          variant,
+          checked,
+        }}
+      />
+      <RadioStyle
+        {...{
+          ...otherProps,
+          disabled,
+          id,
+          defaultChecked,
+          name,
+        }}
+        onFocus={() => {
+          setFocus(true)
```

Comment:
Can't this be solved by CSS alone? Using `:focus` style rule, for instance.

Reply:
I need `focus` as a prop:
```
${({ checked, disabled, focus }) =>
    !checked &&
    !disabled && // IE11 fix. If an element is disabled, you can still hover on it on IE11.
    !focus && // don't override the outline if element is also focussed
    css`
```